File: app/src/main/java/com/example/tim/communityclock/di/BuildersModule.java
package com.example.tim.communityclock.di;

import com.example.tim.communityclock.ui.alarmdisplay.AlarmDisplayActivity;
import com.example.tim.communityclock.ui.alarmdisplay.AlarmDisplayActivityModule;
import com.example.tim.communityclock.ui.main.MainModule;
import com.example.tim.communityclock.ui.main.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class BuildersModule {

    @ContributesAndroidInjector(modules = MainModule.class)
    abstract MainActivity bindMainActivity();

    @ContributesAndroidInjector(modules = AlarmDisplayActivityModule.class)
    abstract AlarmDisplayActivity bindAlarmDisplayActivity();
}

File: app/src/main/java/com/example/tim/communityclock/di/BuildersModule.kt
package com.example.tim.communityclock.di

import com.example.tim.communityclock.ui.alarmdisplay.AlarmDisplayActivity
import com.example.tim.communityclock.ui.main.MainActivity
import com.example.tim.communityclock.ui.setalarm.SetAlarmActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class BuildersModule {

    @ContributesAndroidInjector
    internal abstract fun bindMainActivity(): MainActivity

    @ContributesAndroidInjector
    internal abstract fun bindAlarmDisplayActivity(): AlarmDisplayActivity

    @ContributesAndroidInjector
    internal abstract fun bindSetAlarmActivity(): SetAlarmActivity
}